plugin/trino-hive/src/main/java/io/trino/plugin/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -419,6 +419,40 @@ private static TableStatistics getTableStatistics(
         return result.build();
     }
 
+    private static TableStatistics createEmptyTableStatisticsWithPartitionColumnStatistics(
+            Map<String, ColumnHandle> columns,
+            Map<String, Type> columnTypes,
+            List<HivePartition> partitions)
+    {
+        TableStatistics.Builder result = TableStatistics.builder();
+        for (Map.Entry<String, ColumnHandle> column : columns.entrySet()) {
+            HiveColumnHandle columnHandle = (HiveColumnHandle) column.getValue();
+            if (columnHandle.isPartitionKey()) {
+                result.setColumnStatistics(
+                        columnHandle,
+                        createPartitionColumnStatistics(columnHandle, columnTypes.get(column.getKey()), partitions));
+            }
+        }
+        return result.build();
+    }
+
+    private static ColumnStatistics createPartitionColumnStatistics(HiveColumnHandle column, Type type, List<HivePartition> partitions)
```

Comment:
Move these methods next to the existing `calculatePartitionsRowCount` and add a code comment about why this is useful. For someone without context it is hard to guess why partitioned column stats would useful without row counts.